### PR TITLE
Simplify retrieval of app id

### PR DESCRIPTION
**Pull request by trask**
A couple of things in this PR
* Simplifies code for retrieving app id
* Removes dependency on `httpasyncclient` (retrieving app id now goes over same http client as ingestion)

### CODE REVIEW

File: core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender43.java [hunk 1]
```diff
@@ -36,6 +36,7 @@
 import org.apache.http.client.config.RequestConfig;
 import org.apache.http.client.methods.CloseableHttpResponse;
 import org.apache.http.client.methods.HttpPost;
```
Comment (heyams) [inline core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender43.java:38]: Look like this import is not needed.

File: core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender.java [hunk 2]
```diff
@@ -26,17 +26,19 @@
...
 public interface ApacheSender {
-    final static int DEFAULT_MAX_TOTAL_CONNECTIONS = 200;
-    final static int REQUEST_TIMEOUT_IN_MILLIS = 60000;
-    final static int DEFAULT_MAX_CONNECTIONS_PER_ROUTE = 20;
+    int DEFAULT_MAX_TOTAL_CONNECTIONS = 200;
+    int REQUEST_TIMEOUT_IN_MILLIS = 60000;
+    int DEFAULT_MAX_CONNECTIONS_PER_ROUTE = 20;
 
-    HttpResponse sendPostRequest(HttpPost post) throws IOException, FriendlyException;
+    HttpResponse sendRequest(HttpUriRequest post) throws IOException, FriendlyException;
```
Comment (heyams) [inline core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender.java:41]: (HttpUriRequest request)?

File: core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/TransmissionNetworkOutput.java [hunk 3]
```diff
@@ -164,7 +164,7 @@ public boolean sendSync(Transmission transmission) {
                 // POST the transmission data to the endpoint
                 request = createTransmissionPostRequest(transmission);
                 httpClient.enhanceRequest(request);
-                response = httpClient.sendPostRequest(request);
+                response = httpClient.sendRequest(request);
```
Comment (heyams) [inline core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/TransmissionNetworkOutput.java:167]: I checked HttpPost (HttpEntityEnclosingRequestBase) is not a type of HttpUriRequest, which is HttpRequest, and HttpMessage.  I don't see other changes related to it.. line 165 returns HttpPost? where is the magic?

File: core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender.java [hunk 4]
```diff
@@ -26,17 +26,19 @@
...
+import org.apache.http.client.methods.HttpUriRequest;
 
 /**
  * Created by gupele on 6/4/2015.
  */
 public interface ApacheSender {
-    final static int DEFAULT_MAX_TOTAL_CONNECTIONS = 200;
-    final static int REQUEST_TIMEOUT_IN_MILLIS = 60000;
-    final static int DEFAULT_MAX_CONNECTIONS_PER_ROUTE = 20;
+    int DEFAULT_MAX_TOTAL_CONNECTIONS = 200;
```
[the remark this answers is not in the document]
Comment (trask) [reply core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender.java:37]: all interface fields are static and final, so `final static` was redundant here

File: core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender43.java (hunk 1 above)
Comment (trask) [reply core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender43.java:38]: all unused imports are cleaned up in #1520

File: core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/TransmissionNetworkOutput.java (hunk 3 above)
Comment (trask) [reply core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/TransmissionNetworkOutput.java:167]: HttpPost extends HttpEntityEnclosingRequestBase extends HttpRequestBase implements HttpUriRequest 😄

File: core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender.java (hunk 2 above)
Comment (trask) [reply core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender.java]: done

File: core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender.java [hunk 5]
```diff
@@ -26,17 +26,19 @@
...
+    int REQUEST_TIMEOUT_IN_MILLIS = 60000;
```
[the remark this answers is not in the document]
Comment (trask) [reply core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender.java:38]: same

File: core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender.java [hunk 6]
```diff
@@ -26,17 +26,19 @@
...
+    int DEFAULT_MAX_CONNECTIONS_PER_ROUTE = 20;
```
[the remark this answers is not in the document]
Comment (trask) [reply core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender.java:39]: same

**Review by heyams — APPROVED** [review]

**Review by heyams — COMMENTED** [review]
one last note: should ApacheSender handle the backoff systematically instead of AppIdSupplier handles it individually?  Use the same backoff timer?

**Comment by trask** [comment]
> one last note: should ApacheSender handle the backoff systematically instead of AppIdSupplier handles it individually? Use the same backoff timer?

could be nice in the future, let's see how things look after we move most of the http client stuff to Azure SDK (where we will be using a different http client)

**Pull request merged**